name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/util/chart/ChartToolsManager.java

```diff
@@ -0,0 +1,523 @@
+                        });
+
+        return new Pair<>(tool, action);
+    }
+
+    private void updateButtonImages(ChartTools activeTool)
+    {
+        buttons.forEach(p -> p.getRight().setImageDescriptor(
+                        (p.getLeft() == activeTool ? p.getLeft().getImageON() : p.getLeft().getImageOFF())
+                                        .descriptor()));
+    }
+
+    public void addContextMenu(IMenuManager manager)
+    {
+        createActions().forEach(p -> manager.add(p.getRight()));
+    }
+
+    private static class CrosshairTool extends ChartToolBase
+    {
+        private final TimelineChart chart;
+        private Point p1;
+        private Color color;
+
+        private CrosshairTool(TimelineChart chart, Color color)
+        {
+            this.chart = chart;
+            this.color = color;
+        }
+
+        @Override
+        void onMouseDown(Event e)
+        {
+            if (e.button != 1)
+                return;
+            p1 = new Point(e.x, e.y);
+
+            chart.redraw();
+        }
+
+        @Override
+        void paintControl(PaintEvent e)
+        {
+            if (p1 == null)
+                return;
+
+            drawCrosshair(e, p1);
+        }
+
+        private LocalDate getDateTime(Point p)
+        {
+            return Instant.ofEpochMilli((long) chart.getAxisSet().getXAxis(0).getDataCoordinate(p.x))
+                            .atZone(ZoneId.systemDefault()).toLocalDate();
+        }
+
+        private void drawCrosshairDateTextbox(PaintEvent e, Point p)
+        {
+            LocalDate date = getDateTime(p);
+            String xText = Values.Date.format(date);
+
+            // Add margin to text
+            Point txtXExtend = e.gc.textExtent(xText);
+            txtXExtend.x += 5;
+            txtXExtend.y += 5;
+
+            Point rectPoint = new Point(0, e.height - txtXExtend.y - 2);
+            Point textPoint = new Point(0, e.height - txtXExtend.y + 1);
+
+            // Visual shift vertical of the container
+            if (p.x <= e.width / 2)
+            {
+                rectPoint.x = p.x + 5;
+                textPoint.x = p.x + 8;
+            }
+            else
+            {
+                rectPoint.x = p.x - txtXExtend.x - 5;
+                textPoint.x = p.x - txtXExtend.x - 1;
+            }
+
+            e.gc.setBackground(Colors.brighter(color));
+            e.gc.setForeground(Colors.getTextColor(color));
+
+            e.gc.fillRoundRectangle(rectPoint.x, rectPoint.y, txtXExtend.x, txtXExtend.y, PADDING, PADDING);
+            e.gc.drawText(xText, textPoint.x, textPoint.y, true);
+        }
+
+        private double getYValue(Point p)
+        {
+            return chart.getAxisSet().getYAxis(0).getDataCoordinate(p.y);
+        }
+
+        private void drawCrosshairValueTextbox(PaintEvent e, Point p)
+        {
+            double value = getYValue(p);
+
+            String yText = new DecimalFormat(Values.Quote.pattern()).format(value);
```

Comment:
We should use the pattern that is currently configured for the given chart:

```
chart.getToolTip().getDefaultValueFormat()
```

This is how it currently looks in the performance chart.

<img width="547" alt="Bildschirmfoto 2023-07-31 um 12 51 15" src="https://github.com/portfolio-performance/portfolio/assets/587976/4908b6aa-4f77-41f0-98de-eb75348eb6b6">

Reply:
Yes, I only tested the security and statement of assets charts. I will take note of it.